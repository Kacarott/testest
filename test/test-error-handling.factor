! Copyright 2022 nomennescio

USING: tools.testest math ;
IN: tests

ERROR: custom-error error-message integer-argument ;

: run-tests ( -- )

"System and custom errors tests" describe#{
  "Unexpected system error in user code" it#{
     <{ 1 0 / -> 1 }>
  }#
  "Unexpected custom error in user code" it#{
     <{ "thrown custom error" 1 custom-error -> 1 }>
  }#
  "Expected custom error" it#{
     <{ "thrown custom error" 1 custom-error -> "thrown custom error" 1 custom-error }>
  }#
  "Missing expected custom error" it#{
     <{ 1 -> "thrown custom error" 1 custom-error }>
  }#
}#
;

MAIN: run-tests