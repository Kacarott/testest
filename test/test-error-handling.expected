
<DESCRIBE::>System and custom errors tests

<IT::>Unexpected system error in user code

<ERROR::>Thrown: ERROR:{ division-by-zero { x 1 } }

<COMPLETEDIN::>0.293243 ms

<IT::>Unexpected custom error in user code

<ERROR::>Thrown: ERROR:{ custom-error { error-message "thrown custom error" } {...

<COMPLETEDIN::>0.117949 ms

<IT::>Expected custom error

<PASSED::>Test Passed

<COMPLETEDIN::>0.077905 ms

<IT::>Missing expected custom error

<FAILED::>Test Failed : Expected :<:LF:>THROWN: ERROR:{ custom-error<:LF:>    { error-message "thrown custom error" }<:LF:>    { integer-argument 1 }<:LF:>}<:LF:>But got :<:LF:>1

<COMPLETEDIN::>0.216107 ms

<IT::>Missing expected thrown error

<FAILED::>Test Failed : Expected :<:LF:>THROWN: "thrown error"<:LF:>But got :<:LF:>"thrown error"

<COMPLETEDIN::>0.040166 ms

<IT::>Missing expected thrown custom error

<FAILED::>Test Failed : Expected :<:LF:>THROWN: ERROR:{ custom-error<:LF:>    { error-message "thrown custom error" }<:LF:>    { integer-argument 1 }<:LF:>}<:LF:>But got :<:LF:>ERROR:{ custom-error<:LF:>    { error-message "thrown custom error" }<:LF:>    { integer-argument 1 }<:LF:>}

<COMPLETEDIN::>0.076680 ms

<COMPLETEDIN::>1.040959 ms
